org-roam.el

```diff
@@ -1680,6 +1700,72 @@ command will offer you to create one."
       (when (y-or-n-p "Index file does not exist.  Would you like to create it? ")
         (org-roam-find-file "Index")))))
 
+;;;###autoload
+(defun org-roam-alias-add ()
+  "Add an alias to Org-roam file.
+
+Return added alias."
+  (interactive)
+  (unless org-roam-mode (org-roam-mode))
+  (let ((alias (read-string "Alias: " )))
+    (when (string-empty-p alias)
+      (user-error "Alias can't be empty"))
+    (org-roam--set-global-prop
+     "ROAM_ALIAS"
+     (combine-and-quote-strings
+      (seq-uniq (cons alias
+                      (org-roam--extract-titles-alias)))))
+    (org-roam-db--update-file (buffer-file-name (buffer-base-buffer)))
+    alias))
+
+;;;###autoload
+(defun org-roam-alias-delete ()
+  "Delete an alias from Org-roam file."
+  (interactive)
+  (unless org-roam-mode (org-roam-mode))
+  (if-let ((aliases (org-roam--extract-titles-alias)))
+      (let ((alias (completing-read "Alias: " aliases nil 'require-match)))
+        (org-roam--set-global-prop
+         "ROAM_ALIAS"
+         (combine-and-quote-strings (delete alias aliases)))
+        (org-roam-db--update-file (buffer-file-name (buffer-base-buffer))))
+    (user-error "No aliases to delete")))
+
+(defun org-roam-tag-add ()
+  "Add a tag to Org-roam file.
+
+Return added tag."
+  (interactive)
+  (unless org-roam-mode (org-roam-mode))
+  (let* ((all-tags
```

Comment:
Here I just want to provide completion from existing tags.

Reply:
`(org-roam-db--get-tags)` gives you what you want, and also applies the `distinct` op via SQL, which should be faster.

Reply:
Oh wow, didn't know about this. Really works great and simplifies implementation 🙇